src/components/main-panel/main-panel.tsx

```diff
@@ -21,20 +21,22 @@ export const MainPanel = () => {
         elements,
         focusedElement,
         focusElement,
+        toggleEditPanel,
     } = useBuilder();
     const renderMode = showSidePanel ? "minimised" : "expanded";
 
     // =========================================================================
     // EVENT HANDLERS
     // =========================================================================
     const handleElementCardClick = (element: TElement) => () => {
-        if (focusedElement.element.internalId !== element.internalId) {
+        if (focusedElement?.element?.internalId !== element.internalId) {
             /**
              * TODO: Add check if element is dirty when setting focus
              * Hardcode to false for now
              */
             focusElement(element, false);
         }
+        toggleEditPanel(element, true);
```

Comment:
`toggleEditPanel` also dispatches a `focus-element` action and in this case, the `isDirty` is set to true which is the opposite of what line 37 is doing.

instead of firing twice with `toggleEditPanel`, you should use `togglePanel`

with this change, the `toggleEditPanel` action is not needed at all and can be removed

Reply:
I also don't think the focusedElement check is needed, can we remove it?